sdk/iot/provisioning/samples/src/paho_iot_provisioning_example.c

```diff
@@ -62,15 +62,17 @@ static az_result read_configuration_entry(
   {
     printf("%s\n", hide_value ? "***" : env);
     az_span env_span = az_span_from_str(env);
-    AZ_RETURN_IF_NOT_ENOUGH_CAPACITY(buffer, az_span_length(env_span));
-    *out_value = az_span_copy(buffer, env_span);
+    AZ_RETURN_IF_NOT_ENOUGH_SIZE(buffer, az_span_size(env_span));
+    az_span_copy(buffer, env_span);
+    *out_value = az_span_slice(buffer, 0, az_span_size(env_span));
```

Comment:
Why not just `*out_value = env_span`?

Reply:
Don't we want to copy the data from `getenv` into the passed in buffer az_span?

Doesn't `*out_value = env_span` means the ptr returned from `getenv` (which is a local on the current stack frame) get's returned to the caller. I don't think that is what we want.

```C
  uint8_t buff[20];
  az_span buff_span = AZ_SPAN_FROM_BUFFER(buff);
  int res = putenv("world=ok_now_what");
  (void)res;

  // This prints hello = ok_now_what
  read_configuration_entry("hello", "world", buff_span, &buff_span);

  assert_int_equal(az_span_size(buff_span), 11);

  uint8_t* ptr = az_span_ptr(buff_span);

  ptr[0] = 'a';
  assert_int_equal(ptr[0], 'a');

  // Call putenv to override that memory address.
  res = putenv("world=ok_now_what");
  (void)res;

  assert_int_equal(ptr[0], 'a'); // this fails with arbitrary value -  0xdd != 0x61
```